### ISSUE TITLE
function_body_length to work also on initializers

### ISSUE THREAD

**Issue by lucasdomene** [realm/SwiftLint#3760]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`function_body_length` doesn't work on `init`. Is that intended?
If so, is there another way to enforce the `init` length the same way as the `function_body_length` do?

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

SwiftLint version: 0.45.0
Installation method used: Homebrew
No configuration file (default options)
Which Xcode version are you using: 13.1

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This does not triggers a violation:

override public init(frame: CGRect) {
    
    super.init(frame: frame)
    
    addSubview(titleView)
    titleView.snp.makeConstraints { make in
      make.top.equalToSuperview()
      make.left.equalToSuperview().inset(34)
      make.right.equalToSuperview().priority(999)
    }
    
    addSubview(iconView)
    iconView.snp.makeConstraints { make in
      make.left.equalToSuperview()
      make.centerY.equalTo(titleView)
      make.size.equalTo(CGSize(width: 24, height: 24))
    }

    addSubview(collapsedBackgroundView)
    collapsedBackgroundView.snp.makeConstraints { make in
      make.left.equalTo(titleView)
      make.top.equalTo(titleView.snp.bottom).offset(10)
      make.right.equalToSuperview().priority(999)
      make.bottom.equalToSuperview().inset(ViewSpacing.small.value)
    }
    
    collapsedBackgroundView.addSubview(expandedBackgroundView)
    expandedBackgroundView.snp.makeConstraints { make in
      make.edges.equalToSuperview()
    }
    expandedBackgroundView.isHidden = true
    
    dottedLineView.clipsToBounds = true
    addSubview(dottedLineView)
    dottedLineView.snp.makeConstraints { make in
      make.centerX.equalTo(iconView)
      make.bottom.equalToSuperview()
      make.left.equalToSuperview().priority(999)
      make.right.equalTo(collapsedBackgroundView.snp.left).priority(999)
      make.top.equalTo(iconView.snp.bottom).offset(6)
    }
    
    containerStackView.axis = .vertical
    containerStackView.distribution = .equalSpacing
    collapsedBackgroundView.addSubview(containerStackView)
    containerStackView.snp.makeConstraints { make in
      make.edges.equalToSuperview()
    }
    
    accessoryIconContainerView.addSubview(accessoryIconView)
    accessoryIconView.snp.makeConstraints { make in
      make.size.equalTo(CGSize(width: 24, height: 24))
      make.centerY.equalToSuperview()
      make.left.right.equalToSuperview()
    }
 
    titlesStackView.axis = .vertical
    titlesStackView.distribution = .equalSpacing
    let cardStackView = UIStackView(arrangedSubviews: [entityImageView, titlesStackView, accessoryIconContainerView])
    cardStackView.axis = .horizontal
    cardStackView.spacing = 14
    cardStackView.layoutMargins = UIEdgeInsets(top: ViewSpacing.medium.value,
                                               left: ViewSpacing.medium.value,
                                               bottom: ViewSpacing.medium.value,
                                               right: ViewSpacing.medium.value)
    cardStackView.isLayoutMarginsRelativeArrangement = true
    
    entityImageView.snp.makeConstraints { make in
      entityImageDimensionConstraint = make.width.equalTo(60).constraint
      entityImageAspectRatioConstraint = make.height.equalTo(entityImageView.snp.width).multipliedBy(1.0 / 1.0).priority(.high).constraint
    }

    containerStackView.addArrangedSubview(cardStackView)
    
    divider.backgroundColor = .white
    divider.snp.makeConstraints { make in
      make.height.equalTo(1/UIScreen.main.scale)
    }
    
    containerStackView.addArrangedSubview(divider)
    containerStackView.addArrangedSubview(buttonRowView)
  }

```
